pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -175,7 +176,15 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
       }
 
       if (loadRangeIndex) {
-        _rangeIndex = new RangeIndexReaderImpl(segmentReader.getIndexFor(columnName, ColumnIndexType.RANGE_INDEX));
+        PinotDataBuffer buffer = segmentReader.getIndexFor(columnName, ColumnIndexType.RANGE_INDEX);
+        int version = buffer.getInt(0);
+        if (version == 0) {
+          _rangeIndex = new RangeIndexReaderImpl(buffer);
+        } else if (version == 1) {
+          _rangeIndex = new BitSlicedRangeIndexReader(buffer, metadata);
+        } else {
+          _rangeIndex = null;
```

Comment:
Suggest throwing exception when getting an file of unknown version

Reply:
What about logging that we didn't build the index? I could imagine evolving this in the future and not throwing would keep this forward compatible in mixed version clusters with a more graceful failure mode (just don't use an index we don't currently understand).

Reply:
Yeah, we should also log when we find the index is configured but not possible to be built (e.g. for string columns).
A more graceful failure mode sounds good to me. We might want to log a warning on not loading the unknown index